graylog2-server/src/main/java/org/graylog/events/processor/EventProcessorExecutionJob.java

```diff
@@ -115,9 +114,15 @@ public JobTriggerUpdate execute(JobExecutionContext ctx) throws JobExecutionExce
         try {
             eventProcessorEngine.execute(config.eventDefinitionId(), parameters);
 
+            // TODO: With cron scheduling does it makes sense/is it possible to reliably perform catch up after the
+            //   server has been down for awhile? Relying solely on JobScheduleStrategies.nextTime was not working after
+            //   server restart for cron scheduling.
```

Comment:
This change was made specifically for playing catch up when the server has been offline for awhile and cron scheduled events should have fired in the meantime. After hours of debugging I could not for the life of me figure out where/why this `nextTime` value was being set to `now` for these cron scheduled events, but it was on the second trigger of the job. 

If I had an event scheduled to run every minute looking at the last minute, shut the server off at minute 10, then restarted the server ten minutes later I would get one run of the cron event looking at the correct timeframe of :10-:11, but the next would be for :19-:20. All other 11-12 through 18-19 would be lost. Providing the `to` date to the `nextTime` function for crons worked but I don't think it should have to be used. Could be from mild sleep deprivation due to baby, maybe the weekend will make this clear to me, but if not maybe someone can plz help.

Reply:
@kingzacko1 Can you write a failing test for the problem? That should make it easier to debug and reason about.

Reply:
@bernd just writing the test for the failure (and taking my eyes off the code for a weekend) made me realize my error. I was conflating the `nextTime` and `nextTo` fields in my testing. Properly setting the `nextTo` field for cron scheduled events and leaving `nextTime` alone resolved the issue.